modules/linear_boltzmann_solvers/c_discrete_ordinates_adjoint_solver/lbs_adj_solver.cc

```diff
@@ -172,26 +118,78 @@ DiscreteOrdinatesAdjointSolver::Execute()
   UpdateFieldFunctions();
 }
 
-size_t
-DiscreteOrdinatesAdjointSolver::AddResponseFunction(const std::string& qoi_name,
-                                                    std::shared_ptr<LogicalVolume> logical_volume,
-                                                    std::shared_ptr<ResponseFunction> function)
+double
+DiscreteOrdinatesAdjointSolver::ComputeInnerProduct()
 {
-  // Make the designation
-  ResponseFunctionDesignation qoi_designation(qoi_name, std::move(logical_volume), function);
-  // Make empty subscriber list (will be populated during initialize)
-  std::vector<size_t> cell_rf_subscriptions;
+  double local_integral = 0.0;
+
+  // Material sources
+  for (const auto& cell : grid_ptr_->local_cells)
+  {
+    if (matid_to_src_map_.count(cell.material_id_) == 0) continue; // Skip if no src
+
+    const auto& transport_view = cell_transport_views_[cell.local_id_];
+    const auto& source = matid_to_src_map_[cell.material_id_];
+    const auto& fe_values = unit_cell_matrices_[cell.local_id_];
+
+    for (const auto& group : groups_)
+    {
+      const int g = group.id_;
+      const double Q = source->source_value_g_[g];
+
+      if (Q > 0.0)
+      {
+        const int num_nodes = transport_view.NumNodes();
+        for (int i = 0; i < num_nodes; ++i)
+        {
+          const size_t dof_map = transport_view.MapDOF(i, 0, g); // unknown map
+
+          const double phi = phi_old_local_[dof_map];
+
+          local_integral += Q * phi * fe_values.Vi_vectors[i];
+        } // for node
+      }   // check source value >0
+    }     // for group
+  }       // for cell
+
+  // Point sources
+  for (const auto& point_source : point_sources_)
+    for (const auto& subscriber : point_source.Subscribers())
+    {
+      const auto& cell = grid_ptr_->local_cells[subscriber.cell_local_id];
+      const auto& transport_view = cell_transport_views_[cell.local_id_];
+      const auto& source_strength = point_source.Strength();
+      const auto& shape_values = subscriber.shape_values;
+
+      for (const auto& group : groups_)
+      {
+        const int g = group.id_;
+        const double S = source_strength[g] * subscriber.volume_weight;
+
+        if (S > 0.0)
+        {
+          const int num_nodes = transport_view.NumNodes();
+          for (int i = 0; i < num_nodes; ++i)
+          {
+            const size_t dof_map = transport_view.MapDOF(i, 0, g); // unknown
+                                                                   // map
```

Comment:
Was the `// unknown \nmap` caused by clang-format, because it did not fit? If so, can this be moved in front of the statement?

Reply:
That was there before, I did not add it. Removing the `unknown map` comments as they is redundant for the variable name.